src/main/java/nextstep/courses/domain/service/CourseService.java

```diff
@@ -0,0 +1,32 @@
+package nextstep.courses.domain.service;
+
+import nextstep.courses.domain.Course;
+import nextstep.courses.domain.CourseRepository;
+import nextstep.courses.domain.Session;
+import nextstep.payments.domain.Payment;
+import nextstep.users.domain.NsUser;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Resource;
+
+@Service("courseService")
+public class CourseService {
+
+    @Resource(name = "courseRepository")
+    private CourseRepository courseRepository;
+
+    public void registerSession(Long courseId, Session session, NsUser user, Payment payment) {
+        Course selectedCourse = courseRepository.findById(courseId);
+        if (session.isOpen() && session.isFree()) {
+            addSessionToCourse(session, selectedCourse);
+        }
+        if (session.isOpen() && !session.isFree() && session.isPaymentCorrect(payment)) {
+            addSessionToCourse(session, selectedCourse);
+        }
```

Comment:
```suggestion
        session.register(selectedCourse);
        courseRepository.save(selectedCourse);
```

`session`에서 `sessionStatus`를 가지고 있습니다.
그런의미에서 `session`에서 로직검증을 한다면 좀 더 도메인에 핵심기능이 응집력있게 모일수 있지 않을까요?

Reply:
말씀하신 부분 반영하여, 검증로직을 도메인으로 옮겼습니다!